### DISCUSSION TITLE
What happen on container without acme.sh volume mounted?

### DISCUSSION THREAD

**Discussion by paolo8417** [nginx-proxy/acme-companion#908] [Q&A]
Hi,

I'v create a container without mount the acme.sh volume as specified (my fault). Now the container is alive. What happen now?

**Comment by buchdag** [comment]
Hi. The `acme.sh` volume is used to make `/etc/acme.sh` persistent. This folder is the one that acme.sh uses to store it own state (ACME accounts for instance) and keep track of which certificate has already been issued, when they will need to be renewed, etc.

As long as your container does not go through a stop / start cycle, not having this volume won't affect normal operation.

On the next restart of your container, acme.sh will have its state reset. It will re-create your ACME account (a new one if you're not using Zero SSL) and re-issue all the certificates. This can be an issue with ACME CAs that have rate limits if the container restarts often or if you have a lot of certificates issued from those CAs.

You can get the current content of your container's `/etc/acme.sh` using `docker cp` then copy it back into a volume if you'd like to.
